File: functions/ffmpeg/variants/go/0_flac_get_max_frame_size_8.go
package main

func flacGetMaxFrameSize(blocksize, ch, bps int32) int32 {
    count := int32(16)
    channelSize := ch * ((bps + 14) / 8)
    var dataSize int32
    if ch == 2 {
        dataSize = ((2*bps + 1) * blocksize + 7) / 8
    } else {
        dataSize = (ch * bps * blocksize + 7) / 8
    }
    count += channelSize + dataSize + 2
    return count
}


func main(){
    flacGetMaxFrameSize(0,0,0)
}

File: functions/ffmpeg/variants/c/0_flac_get_max_frame_size_8.c
#include <stdint.h>

static int flac_get_max_frame_size(int blocksize, int ch, int bps) 
{
    int count = 16;
    int bits_per_sample = (bps + 14) / 8;
    count += ch * bits_per_sample;
    
    int frame_payload;
    if (ch == 2) {
        frame_payload = ((2 * bps + 1) * blocksize + 7) / 8;
    } else {
        frame_payload = (ch * bps * blocksize + 7) / 8;
    }
    count += frame_payload;
    
    count += 2;
    
    return count;
}


int main(){
    flac_get_max_frame_size(0,0,0);
}
